src/Pipeline.h

```diff
@@ -121,8 +122,8 @@ class Pipeline {
         }
         template<typename T, typename... Args,
                  typename = typename std::enable_if<Internal::all_are_convertible<Buffer<>, Args...>::value>::type>
-        RealizationArg(Buffer<T> &a, Args &&...args) {
-            buffer_list.reset(new std::vector<Buffer<>>({a, args...}));
+        RealizationArg(Buffer<T> &a, Args &&...args)
+            : buffer_list(std::make_unique<std::vector<Buffer<>>>(std::vector<Buffer<>>{a, std::forward<Args>(args)...})) {
```

Comment:
I thought make_unique forwarded args to a constructor, so can't ```std::vector<Buffer<>>{a, std::forward<Args>(args)...}``` just be ```a, std::forward<Args>(args)...```?

Reply:
You'd think so, but that ends up generating oodles of `no matching constructor for initialization of 'std::__1::vector<Halide::Buffer<void>>'` errors. I'll take another look to see if I can figure out why that isn't working.

Reply:
Actually, no: std::vector doesn't have a ctor that accepts a variadic list of items, which is what that would match to. (It does have a ctor that accepts an initializer list, and of course a copy-constructor, which is what the code above is using.)

Reply:
FYI, the reason that just a plain {wrap-in-braces} here isn't good enough is that a caller passes multiple Buffers of different types (e.g. multiple_scatter.cpp:197) can make type inference problematic here; adding an explicit expectation of the resulting uniform type is necessary. 

That said, using std::initializer_list is likely a better/cheaper way that std::vector here, so I'll do that.